### ISSUE TITLE
Add version number to API

### ISSUE THREAD

**Issue by hrajagers** [csdms/bmi#8]
On behalf of eWaterCycle II

**Comment by mcflugen** [comment]
We should use [Semantic Versioning](https://semver.org/).

**Comment by mdpiper** [comment]
@hrajagers I'm beginning to address this; see, for example, the tagged version [v2.0b0](https://github.com/csdms/bmi/releases/tag/v2.0b0), where the SIDL file header now has the version:

```java
//
// The Basic Model Interface (BMI)
//
package csdms version 2.0b0 {
  interface bmi {

...
```

**Comment by hrajagers** [comment]
Hi Mark,


Thank you. With the issue I intended to ask for a routine/function as part
of the BMI API that would allow me to query the version of BMI implemented
by a component such that the caller could adjust to the BMI version
supported by the component. I'm not sure that the way in which the version
number is added now can be used for this purpose across language bindings.

**Comment by mdpiper** [comment]
@hrajagers Got it. Thanks, Bert!

**Comment by jdhughes-usgs** [comment]
@mdpiper we have also been looking at implementing a function for a model to return the model version through the MODFLOW API but this might be something that would be generally useful in BMI. This became an issue for us when users were using an outdated version of the MODFLOW dll/dylib/so.

**Comment by mcflugen** [comment]
I suggest a slight modification,. What do you think of,
```c
int get_bmi_version(out string version);
```
To me, `get_version` would refer to the version of the model itself, not the version of BMI it implements.

If, in the future, we want a function to return the model version, we could go with `get_version` or `get_component_version`. I suspect this is something we will want in the future, particularly if we add an extension for serializing/deserializing models.

**Comment by mdpiper** [comment]
Resolved with #111.
